examples/spring/src/main/kotlin/com/expediagroup/graphql/examples/query/OrganizationQuery.kt

```diff
@@ -0,0 +1,54 @@
+package com.expediagroup.graphql.examples.query
+
+import com.expediagroup.graphql.annotations.GraphQLDescription
+import com.expediagroup.graphql.examples.model.Company
+import com.expediagroup.graphql.examples.model.Employee
+import com.expediagroup.graphql.spring.operations.Query
+import graphql.schema.DataFetcher
+import graphql.schema.DataFetchingEnvironment
+import org.springframework.beans.factory.BeanFactory
+import org.springframework.beans.factory.BeanFactoryAware
+import org.springframework.context.annotation.Scope
+import org.springframework.stereotype.Component
+import java.util.concurrent.CompletableFuture
+
+@Component
+class CompanyService {
+    private val companies = listOf(
+        Company(id = 1, name = "FirstCompany"),
+        Company(id = 2, name = "SecondCompany")
+    )
+
+    fun getCompanies(ids: List<Int>): List<Company> = companies
+}
+
+@Component
+class EmployeeQuery : Query {
+    private val employees = listOf(
+        Employee(name = "Mike", companyId = 1),
+        Employee(name = "John", companyId = 1),
+        Employee(name = "Steve", companyId = 2)
+    )
+
+    @GraphQLDescription("Get all employees")
+    fun employees(): List<Employee> {
+        return employees
+    }
+}
+
+@Component("CompanyDataFetcher")
+@Scope("prototype")
+class CompanyDataFetcher : DataFetcher<CompletableFuture<Company>>, BeanFactoryAware {
```

Comment:
@smyrick hey. Am I understand that correctly that this thing will be called anytime somebody queries the` company` field `company`
`lateinit var Company` on Employee and ONLY the Employee, and not other classes?
Or I have to branch here or whatever?

Maybe some drawing or ASCII or draw.io graph will help me understand the connection between beans and the whole resolve cycle as I'm unsure :)

Reply:
@RIP21 The DataFetcher is the name of the "resolver" for graphql-java. You can see the docs here: https://www.graphql-java.com/documentation/v16/data-fetching/

This is a data fetcher for the entire `Company` class. GraphQL Kotlin normally would handle creating the different data fetchers for you, but in this case the user wanted have a `DataLoader` which is a separate GraphQL specific caching mechanism. The `DataFetcher` is invoked when ever there is a field that returns a `Company` and, using `DataLoader`s, they are cache the company info for each employee so they don't have to make multiple calls to get the the duplicate company data for all the employees